api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java

```diff
@@ -31,7 +31,7 @@
  * <p>A ConfigSource will get picked up via the
  * {@link java.util.ServiceLoader} mechanism and must get registered via
  * META-INF/services/javax.config.spi.ConfigSource</p>
- * The other custom config source can be added programmatically via {@link org.eclipse.microprofile.config.ConfigProvider}.
+ * The other custom config source can be added programmatically via {@link org.eclipse.microprofile.config.ConfigProvider}. << How?
```

Comment:
ConfigProvider.getConfig() is managed and always return the same one for the same classloader. ConfigProvider.getBuilder().withSoruces(...) creates a new instance each time, hence unmanaged. If you call ConfigProvider.setConfig(), which will store the config. Later on, when you call ConfigProvider.getConfig(), which will return you the one just stored. Remember, setConfig() is only successful if there is no config for this classloadder was stored. Otherwise, you will get exception. ConfigProvider.releaseConfig() will release the config.